matlab_simulation/04-sensor/scanner_bresenham/robot_motion.m

```diff
@@ -0,0 +1,108 @@
+function [X] = robot_motion( map, X, x0, T)
```

Comment:
I'm wondering if this function and "collision_pred" should be combined? I didn't dig in too much but it looks like they do a lot of similar logic. Just a thought.